### PR TITLE
[MPLUGIN-382] exclude dependency in provided scope from plugin descriptor

### PR DESCRIPTION
**Pull request by slawekjaranowski**
`plugin.xml` descriptor - should only contain dependencies in compile scope

### CODE REVIEW

**Comment by michael-o** [comment]
@cstamas Can you have a look?

File: maven-plugin-tools-generators/src/main/java/org/apache/maven/tools/plugin/generator/GeneratorUtils.java [hunk 1]
```diff
@@ -122,14 +122,19 @@ public static void element( XMLWriter w, String name, String value, boolean asTe
...
-     * @return list of component dependencies
+     * @return list of component dependencies, only in compile scope
      */
     public static List<ComponentDependency> toComponentDependencies( Collection<Artifact> artifacts )
     {
         List<ComponentDependency> componentDeps = new LinkedList<>();
 
         for ( Artifact artifact : artifacts )
         {
+            if ( !Artifact.SCOPE_COMPILE.equals( artifact.getScope() ) )
```
Comment (cstamas) [inline maven-plugin-tools-generators/src/main/java/org/apache/maven/tools/plugin/generator/GeneratorUtils.java:133]: I would rather make `SCOPE = PROVIDED` (then continue). What about runtime scope? Just to be on safe side.

Comment (slawekjaranowski) [reply maven-plugin-tools-generators/src/main/java/org/apache/maven/tools/plugin/generator/GeneratorUtils.java]: fixed

**Review by cstamas — APPROVED** [review]

**Comment by jira-importer** [comment]
Resolve #692

**Pull request merged**